### ISSUE TITLE
Avoid trivial denial-of-service opportunities

### ISSUE THREAD

**Issue by arnetheduck** [status-im/nim-eth-p2p#61]
There's plenty of code like the following, that every time creates a security problem due to how one can easily stuff the client with silent peers, as a form of cheap DoS - without a timeout they will stay in limbo indefinitely:

https://github.com/status-im/nim-eth-p2p/blob/53e98faa74ce3f9e690fc985e56d18cf76b67b42/eth_p2p/rlpx.nim#L1378

```
Jacek Sieka
@arnetheduck
18:11

I'm looking at the peer code a bit:

    await transport.readExactly(addr authMsg[0], len(authMsg))

if the peer doesn't send anything, what will happen?
(this is from the incoming-peer handler,rlpxAccept
Eugene Kabanov
@cheatfate
18:15
@arnetheduck this depends, if peer doesn't send anything but not disconnect then this function will stuck, if it disconnects and number of bytes transferred will be less then len(authMsg) TransportIncompleteError exception will be raised, in rare cases TransportOsError can be raised.
```

**Comment by kdeme** [comment]
Yes, I've seen similar issues in even more obvious places: like [here](https://github.com/status-im/nim-eth-p2p/blob/master/eth_p2p/rlpx_protocols/les_protocol.nim#L210) and [also here](https://github.com/status-im/nim-eth-p2p/blob/master/eth_p2p/rlpx_protocols/eth_protocol.nim#L55).
It's also in the documentation: https://github.com/status-im/nim-eth-p2p/blob/master/README.md#implementing-handshakes-and-reacting-to-other-events 

I had initially implemented it like that in Whisper, but then temporary fixed like [this](https://github.com/status-im/nim-eth-p2p/blob/1e5eeec6bb65f0f6c2cbe4039d05f1ba5f407970/eth_p2p/rlpx_protocols/shh_protocol.nim#L691), until zah added the [`handshake`](https://github.com/status-im/nim-eth-p2p/blob/master/eth_p2p/rlpx.nim#L1141).

Should be altered in `eth`, `les` and documentation also.
